source/extensions/tracers/xray/tracer.h

```diff
@@ -156,6 +163,11 @@ class Span : public Tracing::Span, Logger::Loggable<Logger::Id::config> {
    */
   const std::string& direction() const { return direction_; }
 
+  /**
+   * Gets this Span's type.
+   */
+  const std::string& type() const { return type_; }
```

Comment:
I also considered using a boolean isChild to mark a given span, but this works too.

Reply:
I think what you have is more flexible.